kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -123,6 +128,8 @@ def exec_kompile(
     if debug_build:
         optimization = 0
 
+    ccopts = list(ccopts) + lib_ccopts(kdist.get('evm-semantics.plugin'), debug_build=debug_build)
```

Comment:
There are thee occurrences of the following pattern in this repo (and there will be two in `kontrol` if this is merged):

```python3
ccopts = ccopts + lib_ccopts(plugin_dir)
kevm_kompile(
    ...,
    ccopts=ccopts,
    ...,
)
```

Furhtermore, this is the only way `kevm_kompile` is called.

What's the advantage of pulling out this detail?

Reply:
Because it brings semantics-specific things (like the fact that there is a plugin, and the location of it) _out_ of `kevm_kompile`, so that the same method can be upstreamed and using by, for example, the MIR semantics.

Reply:
In that case, consider bringing out the semantic independent details instead by extracting a function:

```python3
def kevm_kompile(..., plugin_dir, ccopts, ...):
    ccopts = list(ccopts) + _lib_ccopts(plugin_dir)
    return generic_kompile(..., ccopts, ...)

def generic_kompile(..., ccopts, ...):
    ...
```

Reply:
Done.